src/main/java/seedu/address/model/person/StudentId.java

```diff
@@ -0,0 +1,61 @@
+package seedu.address.model.person;
+
+import static java.util.Objects.requireNonNull;
+import static seedu.address.commons.util.AppUtil.checkArgument;
+
+/**
+ * Represents a Person's StudentId in the address book.
+ * Guarantees: immutable; is valid as declared in {@link #isValidStudentId(String)}
+ */
+public class StudentId {
+
+
+    public static final String MESSAGE_CONSTRAINTS =
+            "Student ID should only contain alphanumeric characters, and it should be 9 characters long";
+    public static final String VALIDATION_REGEX = "[a-zA-Z]\\d{7}[a-zA-Z]";
+    public final String value;
+
+    /**
+     * Constructs a {@code StudentId}.
+     *
+     * @param studentId A valid studentId.
+     */
+    public StudentId(String studentId) {
+        requireNonNull(studentId);
```

Comment:
RequireNonNull will throw a null pointer exception if the argument is null. Have you accounted for the handling of this null pointer exception? Perhaps assertion statement could be a plausible solution if you want to stop execution of the program.